tests/torch/quantization/test_range_init.py

```diff
@@ -671,53 +812,66 @@ def range_init_call_count_test_struct(request):
+    mocker.patch(
+        "nncf.torch.quantization.init_range.get_median_mad_statistic_collector", new=range_threesigma_init_create_spy
+    )
+
     config = create_config()
     config["compression"]["initializer"]["range"] = range_init_call_count_test_struct.range_init_config
     data_loader = TestRangeInit.create_dataloader(True, config, 10)
     config.register_extra_structs([QuantizationRangeInitArgs(data_loader)])
 
-    range_minmax_init_create_spy = mocker.spy(PTMinMaxStatisticCollector, "__init__")
-    range_meanminmax_init_create_spy = mocker.spy(PTMeanMinMaxStatisticCollector, "__init__")
-    range_threesigma_init_create_spy = mocker.spy(PTMedianMADStatisticCollector, "__init__")
-
-    range_minmax_init_register_input_spy = mocker.spy(PTMinMaxStatisticCollector, "_register_input")
-    range_meanminmax_init_register_input_spy = mocker.spy(PTMeanMinMaxStatisticCollector, "_register_input")
-    range_threesigma_init_register_input_spy = mocker.spy(PTMedianMADStatisticCollector, "_register_input")
-
     TestRangeInit.create_algo_and_compressed_model(config)
 
-    assert (
-        range_minmax_init_create_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_initializer_create["min_max"]
-    )
-    assert (
-        range_meanminmax_init_create_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_initializer_create["mean_min_max"]
-    )
-    assert (
-        range_threesigma_init_create_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_initializer_create["three_sigma"]
-    )
-
-    assert (
-        range_minmax_init_register_input_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_register_input["min_max"]
-    )
-    assert (
-        range_meanminmax_init_register_input_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_register_input["mean_min_max"]
-    )
-    assert (
-        range_threesigma_init_register_input_spy.call_count
-        == range_init_call_count_test_struct.expected_call_count_register_input["three_sigma"]
-    )
-
-
-QUANTIZER_RANGE_INITIALIZERS = ["min_max", "threesigma", "mean_min_max", "percentile", "mixed_min_max"]
+    for stat_type, spy in [
+        ("min_max", range_minmax_init_create_spy),
+        ("mean_min_max", range_meanminmax_init_create_spy),
+        ("three_sigma", range_threesigma_init_create_spy),
+    ]:
+        assert spy.call_count == range_init_call_count_test_struct.expected_call_count_initializer_create[stat_type]
+        collected_samples = 0
+        for tensor_collector in spy.return_values_list:
+            cur_values = set()
+            for aggr in tensor_collector.aggregators.values():
+                cur_values.add(aggr._collected_samples)
+            assert len(cur_values) == 1
+            collected_samples += cur_values.pop()
```

Comment:
It counts per layer range init collectors are called the required number of times. Should I add docstring, comments and type hints to make it clearer?

Reply:
I mean that the test only concerns itself with the call count. It may be excessive to inspect the samples themselves in the same case, and instead do this kind of testing elsewhere.

Reply:
This test checks per layer range init collectors are called the required number of times, test `test_init_ranges_are_set` inspects samples themself. This test structure were present before. What is the problem?

Reply:
Ok, looks like I misunderstood the code (which is, by the way, consequence of bad naming, on my own account since I authored that code). I thought that `aggr._collected_samples` (which is, by the way, also a violation of encapsulation) would be containing the collected samples themselves, when in fact it is the *number* of collected samples. Should rename this to `num_collected_samples` or something like that.